presto-main/src/main/java/com/facebook/presto/operator/MultiChannelGroupByHash.java

```diff
@@ -173,10 +205,21 @@ public void addPage(Page page)
             return;
         }
 
+        // assure there is only 1 column if nullChannel is present
+        if (nullChannel.isPresent()) {
+            checkState(channels.length == 1, "channel length is not 1 while nullChannel is present");
+        }
+
         // get the group id for each position
         int positionCount = page.getPositionCount();
         for (int position = 0; position < positionCount; position++) {
-            // get the group for the current row
+            if (nullChannel.isPresent()) {
+                boolean isIndterminate = page.getBlock(nullChannel.get()).isNull(position);
+                if (isIndterminate) {
+                    nullRows.add(ImmutableList.of((Block) types.get(0).getObject(page.getBlock(0), position)));
```

Comment:
This should probably be "indeterminateRows" instead of "nullRows".

In any case, we should probably be storing positions instead of materializing all the indeterminate row.